pages/mesosphere/dcos/1.13/installing/production/advanced-configuration/configuration-reference/index.md

```diff
@@ -251,35 +251,41 @@ Use this to set up a custom CA certificate. See [using a Custom CA Certificate](
 Use this to set up a custom CA certificate. See [using a Custom CA Certificate](/mesosphere/dcos/1.13/security/ent/tls-ssl/ca-custom#configuration-parameter-reference) page for a detailed configuration parameter reference. 
 
 ### cluster_docker_credentials
-
 The dictionary of Docker credentials to pass.
 
-- If unset, a default empty credentials file is created at `/etc/mesosphere/docker_credentials` during DC/OS install. A sysadmin can change credentials as needed. A `systemctl restart dcos-mesos-slave` or `systemctl restart dcos-mesos-slave-public` is required for changes to take effect.
-- You can also specify by using the `--docker_config` JSON [format](http://mesos.apache.org/documentation/latest/configuration/). You can write it as YAML in the `config.yaml` file and it will automatically be mapped to the JSON format for you. This stores the Docker credentials in the same location as the DC/OS internal configuration (`/opt/mesosphere`). If you need to update or change the configuration, you will have to create a new DC/OS internal configuration.
+- Creates a credentials file at `cluster_docker_credentials_path` during DC/OS install. A sysadmin can change credentials stored in that file as needed as long as `cluster_docker_credentials_dcos_owned` is set to `'false'`. A `systemctl restart dcos-mesos-slave` or       `systemctl restart dcos-mesos-slave-public` is required for changes to the file to take effect.
+- You can also specify this value by using the `--docker_config` JSON [format](http://mesos.apache.org/documentation/latest/configuration/). You can write it as YAML in the `config.yaml` file and it will automatically be mapped to the JSON format for you.
 
 **Note:**
-- `cluster_docker_credentials` takes effect only when [`cluster_docker_credentials_enabled`](#cluster-docker-credentials-enabled) is set to `'true'`
-- `cluster_docker_credentials` takes effect during an upgrade only when `cluster_docker_credentials_dcos_owned` is set to `'true'`.
+- `cluster_docker_credentials` takes effect only when [`cluster_docker_credentials_enabled`](#cluster-docker-credentials-enabled) is set to `'true'`.
+- `cluster_docker_credentials` must not be left unset when `cluster_docker_credentials_enabled` is set to `'true'`. At a minimum it must be set to a blank configuration: `'{}'`
+- `cluster_docker_credentials` takes effect during an install only when either `cluster_docker_credentials_dcos_owned` or `cluster_docker_credentials_write_to_etc` are set to `'true'`.
```

Comment:
I see you removed the description of `cluster_docker_credentials_write_to_etc`, right? Can you please elaborate a bit why? I guess we still need a description for it, otherwise it may be hard for use to know what `cluster_docker_credentials_write_to_etc` actually mean.

Reply:
That was removed accidentally. I will update this PR shortly to add a description for it.

>  I guess we still need a description for it, otherwise it may be hard for use to know what cluster_docker_credentials_write_to_etc actually means

Agreed. Although it's already quite hard to know what is means since it doesn't seem to behave as the current docs suggest and I just now discovered a bug with these parameters too: COPS-6402

Reply:
@qianzhangxa `cluster_docker_credentials_write_to_etc` has been added back in.

Reply:
Thanks @jeffwhite530 !

Reply:
> " ... if both cluster_docker_credentials_dcos_owned [and] cluster_docker_credentials_write_to_etc are set to false but cluster_docker_credentials_enabled is set to true ..."

@qianzhangxa , in this case, DC/OS will not create the file for you. The agent will fail to start until the file is created and a valid configuration is placed into it. This would be useful if you have an outside process such as Ansible which manages the file for you. IMHO a better configuration is one that creates the file for you but does not ever overwrite an existing file which Ansible or whatever placed there. That's already documented here:

https://docs.d2iq.com/mesosphere/dcos/2.1/deploying-services/private-docker-registry/#using-cluster-docker-credentials-to-set-cluster-wide-registry-credentials

... note that the page has this change pending: https://github.com/mesosphere/dcos-docs-site/pull/3130